custom_components/plugwise/binary_sensor.py

```diff
@@ -184,32 +182,47 @@ async def async_setup_entry_gateway(hass, config_entry, async_add_entities):
     async_add_entities(entities, True)
 
 
-class GwBinarySensor(SmileSensor, BinarySensorEntity):
-    """Representation of a Plugwise binary_sensor."""
-
-    def __init__(
-        self, api, coordinator, name, dev_id, enabled_default, binary_sensor, model
-    ):
-        """Set up the Plugwise API."""
-        self._enabled_default = enabled_default
+class SmileBinarySensor(SmileGateway):
+    """Represent Smile Binary Sensors."""
 
-        super().__init__(
-            api, coordinator, name, dev_id, self._enabled_default, binary_sensor
-        )
+    def __init__(self, api, coordinator, name, dev_id, binary_sensor):
+        """Initialise the binary_sensor."""
+        super().__init__(api, coordinator, name, dev_id)
 
         self._binary_sensor = binary_sensor
 
-        self._is_on = False
         self._icon = None
-        self._state = None
+        self._is_on = False
+
+        if dev_id == self._api.heater_id:
+            self._entity_name = "Auxiliary"
+
+        sensorname = binary_sensor.replace("_", " ").title()
+        self._name = f"{self._entity_name} {sensorname}"
+
+        if dev_id == self._api.gateway_id:
+            self._entity_name = f"Smile {self._entity_name}"
 
         self._unique_id = f"{dev_id}-{binary_sensor}"
 
+    @property
+    def icon(self):
+        """Return the icon of this entity."""
+        return self._icon
+
     @property
     def is_on(self):
         """Return true if the binary sensor is on."""
         return self._is_on
 
+    @callback
+    def _async_process_data(self):
+        """Update the entity."""
+        raise NotImplementedError
+
+class GwBinarySensor(SmileSensor, BinarySensorEntity):
```

Comment:
Shouldn't this be `...(SmileBinarySensor, ...):`?

Reply:
Yup